crypto/rand/rand_lib.c

```diff
@@ -12,27 +12,50 @@
 #include "internal/cryptlib.h"
 #include <openssl/opensslconf.h>
 #include "internal/rand.h"
-
 #include <openssl/engine.h>
+#include "internal/thread_once.h"
 
 #ifndef OPENSSL_NO_ENGINE
 /* non-NULL if default_RAND_meth is ENGINE-provided */
 static ENGINE *funct_ref = NULL;
+static CRYPTO_RWLOCK *rand_engine_lock = NULL;
 #endif
 static const RAND_METHOD *default_RAND_meth = NULL;
```

Comment:
In other projects I usually see a comment "requires rand_meth_lock for access" or semantically similar; do we want to do something like that?